### ISSUE TITLE
[BUG] [DART] Using multipart/form-data breaks on File.fromJson

### ISSUE THREAD

**Issue by freshteapot** [OpenAPITools/openapi-generator#7589]
#### Bug Report Checklist

- [ ] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [ ] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [ ] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

The generated code.

```dart
  HttpAssetUploadRequestBody.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    sharedWith = json['shared_with'];
    file = (json['file'] == null) ?
      null :
      File.fromJson(json['file']);
  }
```
The problem "fromJson" is not a method on class "File", Also "file" is  "MultipartFile", which possibly means "File" should be "MultipartFile". (I am sorry for file and File).

- https://api.dart.dev/stable/2.10.0/dart-io/File-class.html
- https://pub.dev/documentation/http/latest/http/MultipartFile-class.html

##### openapi-generator version

```sh
openapi-generator version
4.3.1
```

##### OpenAPI declaration file content or url
It should fail with the petstore yaml file

Your more than welcome to try against

https://gist.github.com/freshteapot/2e33445c0a2f88220a156e1f61589590

##### Generation Details
From what I can understand using the petstore example will trigger this to happen.

Working with https://gist.github.com/freshteapot/2e33445c0a2f88220a156e1f61589590.
Use

```sh
DART_POST_PROCESS_FILE="/usr/local/bin/dartfmt -w" \
openapi-generator generate -i /tmp/openapi/one/learnalist.yaml -g dart -o /tmp/openapi/dart \
--additional-properties ensureUniqueParams=false
```

##### Related issues/PRs

Will dig up things that remotely link.

##### Suggest a fix

Maybe not using fromJson and just treat "file" as filename.

file, in this case is "MultipartFile". The below code compiles, not 100% sure if it works, as I haven't used the new api in my flutter app.

```dart
HttpAssetUploadRequestBody.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    sharedWith = json['shared_with'];
    file = (json['file'] == null)
        ? null
        : MultipartFile.fromString("file", json['file']);
  }
```

Possibly the issue is linked to https://github.com/OpenAPITools/openapi-generator/blob/3f18d0fd102a92bcbd7edc20b592741fc2da96c0/modules/openapi-generator/src/main/resources/dart2/class.mustache#L73-L75.

I suspect this might also effect toJson, as well. But at least it compiles.

**Comment by kuhnroyal** [comment]
I fixed something related to this in 5.0.0 - can you test this again?

**Comment by sergiofigueras** [comment]
Hello there @kuhnroyal ,

I'm facing the same issue.

```
Target dart2js failed: Exception: ../../development/flutter/.pub-cache/git/estant-rest-client-dart-71ddec93c90e3d77411ee872e464f3cb68e1ab5a/lib/model/file.dart:54:21:
Error: Method not found: 'MultipartFile.fromJson'.
      MultipartFile.fromJson(json['absoluteFile']);
                    ^^^^^^^^
../../development/flutter/.pub-cache/git/estant-rest-client-dart-71ddec93c90e3d77411ee872e464f3cb68e1ab5a/lib/model/file.dart:58:21:
Error: Method not found: 'MultipartFile.fromJson'.
      MultipartFile.fromJson(json['canonicalFile']);
                    ^^^^^^^^
../../development/flutter/.pub-cache/git/estant-rest-client-dart-71ddec93c90e3d77411ee872e464f3cb68e1ab5a/lib/model/file.dart:70:21:
Error: Method not found: 'MultipartFile.fromJson'.
      MultipartFile.fromJson(json['parentFile']);
                    ^^^^^^^^
../../development/flutter/.pub-cache/git/estant-rest-client-dart-71ddec93c90e3d77411ee872e464f3cb68e1ab5a/lib/model/resource.dart:32:21:
Error: Method not found: 'MultipartFile.fromJson'.
      MultipartFile.fromJson(json['file']);
                    ^^^^^^^^
Error: Compilation failed.
```

Currently, I'm using 5.0.0.

```
➜  estant-rest-client-dart git:(master) openapi-generator version
5.0.0
```

Is there anything else that you need?

Thanks.

**Comment by sergiofigueras** [comment]
Hello there @kuhnroyal ,

I had the same issue in 4.3.1 , but its now fixed in 5.0.0 .

Thanks!

**Comment by kuhnroyal** [comment]
@wing328 This can be closed
